
<!DOCTYPE html>
<html lang='en'>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <title>Tree - rpms/plasma-workspace - src.fedoraproject.org</title>
        <link rel="shortcut icon" type="image/vnd.microsoft.icon"
        href="/theme/static/favicon.ico?version=5.13.3"/>
    <link href="/theme/static/fedora-bootstrap-1.3.0/fedora-bootstrap.min.css?version=5.13.3"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/fonts/fonts.css?version=5.13.3"
        rel="stylesheet" type="text/css" />
    <link href="/theme/static/fonts/hack_fonts/css/hack-extended.min.css?version=5.13.3"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/theme.css?version=5.13.3"
        type="text/css" rel="stylesheet" />

    <link type="text/css" rel="stylesheet" nonce="bYHSDeYpamaMeW4MROg2tpoFA" href="/static/vendor/font-awesome/font-awesome.css?version=5.13.3"/>
    <link type="text/css" rel="stylesheet" nonce="bYHSDeYpamaMeW4MROg2tpoFA" href="/static/pagure.css?version=5.13.3"/>
<link nonce="bYHSDeYpamaMeW4MROg2tpoFA" rel="stylesheet" href="/static/vendor/highlight.js/styles/github.css?version=5.13.3"/>
<link nonce="bYHSDeYpamaMeW4MROg2tpoFA" rel="stylesheet" href="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.css?version=5.13.3"/>
<style nonce="bYHSDeYpamaMeW4MROg2tpoFA">
  .hljs {
    background: #fff;
  }
</style>
  </head>
  <body id="home">
    
    <!-- start masthead -->
    <nav class="navbar navbar-light masthead p-0 navbar-expand">
      <div class="container">
        <a href="/" class="navbar-brand">
        <img height="40" src="/theme/static/pagure-logo.png?version=5.13.3"
             alt="pagure Logo" id="pagureLogo"/>
        </a>
        <ul class="navbar-nav ml-auto">

          

          <li class="nav-item">
            <a class="btn btn-primary" href="/login/?next=https://src.fedoraproject.org/rpms/plasma-workspace/blob/rawhide/f/disable-toplevel-fixed-positions-on-wayland.patch">Log In</a>
          </li>
        </ul>
      </div>
    </nav>
    <!-- close masthead-->

    <div class="bodycontent">


<div class="bg-light border border-bottom pt-3">
  <div class="container">
    <div class="row mb-3">
      <div class="col-6">
        <div class="row">
        <div class="col-auto pr-0">
            <h3>
<i class="fa fa-archive text-muted"></i></h3>
        </div>
        <div class="col-auto pl-2">
            <h3 class="mb-0">
<a href="/projects/rpms/%2A">rpms</a>&nbsp;/&nbsp;<a href="/rpms/plasma-workspace"><strong>plasma-workspace</strong></a>
            </h3>
        </div>
        </div>
      </div>
      <div class="col-6 text-right">
        <div class="btn-group">
        <div class="btn-group">
        <a href="#"
            class="btn btn-sm dropdown-toggle btn-outline-primary"
            data-toggle="dropdown" id="watch-button">
          <i class="fa fa-clone fa-fw"></i>
          <span>Clone</span>
        </a>
        <div class="dropdown-menu dropdown-menu-right">
          <div class="m-3" id="source-dropdown">
            <div>
              <h5><strong>Source Code</strong></h5>

              <div class="form-group">
                <div class="input-group input-group-sm">
                  <div class="input-group-prepend"><span class="input-group-text">GIT</span></div>
                  <input class="form-control bg-white select-on-focus" type="text" value="https://src.fedoraproject.org/rpms/plasma-workspace.git" readonly>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  </div>
</div>

<ul class="nav nav-tabs nav-small border-bottom-0">
  <li class="nav-item mr-2 text-dark">
    <a class="nav-link active" href="/rpms/plasma-workspace">
        <i class="fa fa-code fa-fw text-muted"></i>
        <span class="d-none d-md-inline">Source</span>
    </a>
  </li>

    <li class="nav-item mr-2 text-dark">
        <a class="nav-link" href="https://bugzilla.redhat.com/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&classification=Fedora&product=Fedora&product=Fedora EPEL&component=plasma-workspace">
          <i class="fa fa-fw text-muted fa-exclamation-circle"></i>
          <span class="d-none d-md-inline">Issues&nbsp;</span>
          <span class="fa fa-external-link"></span>
        </a>
    </li>

    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/plasma-workspace/pull-requests">
          <i class="fa fa-fw text-muted fa-arrow-circle-down"></i>
          <span class="d-none d-md-inline">Pull Requests&nbsp;</span>
          <span class="badge badge-secondary py-0 d-none d-md-inline">
            0
          </span>
      </a>
    </li>


    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/plasma-workspace/stats">
          <i class="fa fa-line-chart fa-fw text-muted"></i>
          <span class="d-none d-md-inline">Stats</span>
      </a>
    </li>


</ul>
  </div>
</div>

<div class="container pt-5 repo-body-container">
  <div class="row">
    <div class="col-2">
<nav class="nav nav-tabs nav-sidetabs flex-column">
  <a class=
      "nav-link nowrap
"
      href="/rpms/plasma-workspace">
      <i class="fa fa-home text-muted fa-fw"></i>&nbsp;<span class="d-none d-md-inline">Overview</span>
  </a>
  <a class=
    "nav-link nowrap
 active"
    href="/rpms/plasma-workspace/tree/rawhide">
    <i class="fa fa-file-code-o text-muted fa-fw"></i>&nbsp;Files
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/plasma-workspace/commits/rawhide">
    <i class="fa fa-list-alt text-muted fa-fw" data-glyph="spreadsheet"></i>&nbsp;Commits
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/plasma-workspace/branches?branchname=rawhide">
    <i class="fa fa-random text-muted fa-fw"></i>&nbsp;Branches
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/plasma-workspace/forks">
    <i class="fa fa-code-fork text-muted fa-fw"></i>&nbsp;Forks
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/plasma-workspace/releases">
    <i class="fa fa-tags text-muted fa-fw"></i>&nbsp;Releases
  </a>

  <div class="col-xs-2 line-height-1"></div>
  <h6>Monitoring status:</h6>
  <div class="btn-group">
    <button title="Monitoring status" class="btn btn-sm btn-outline-primary disabled"
        id="monitoring-button">
      <i id="monitoring-icon" class="fa fa-fw fa-eye"></i>
      <span id="monitoring-label" class="fa fa-circle-o-notch fa-spin fa-1x fa-fw"></span>
    </button>
  </div>

  <div class="col-xs-2 line-height-1"></div>
  <div id="orphan-section" class="pt-3">
      <div class="col-xs-2 line-height-1"></div>

  </div>

  <div class="pt-3">
    <div class="col-xs-2 line-height-1">
    <h6>Bugzilla Assignee:</h6>
      <dl>
        <dt>Fedora: </dt>
        <dd id="fedora_assignee_txt">
        @kde-sig
        </dd>
        <dt>EPEL: </dt>
        <dd id="epel_assignee_txt">
        dvratil
        </dd>
      </dl>
    </div>
  </div>

  <div class="modal fade" id="modal_assignee" tabindex="-1"
          role="dialog" aria-labelledby="Bugzilla assignee" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Bugzilla Assignee</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
        <form id="change_assignees">
          <div class="modal-body">
            <label for="fedora_assignee">Fedora</label>
            <input title="Default assignee for Fedora in bugzilla - Empty input resets to default"
              class="form-control" name="fedora_assignee" id="fedora_assignee" value="@kde-sig"/>
            <label for="epel_assignee">EPEL</label>
            <input title="Default assignee for EPEL in bugzilla (if applicable) - Empty input resets to default"
              class="form-control" name="epel_assignee" id="epel_assignee" value="dvratil" />
            <p class="pt-2">
              These two fields allow to specify a different default assignee for ticket opened against
              this package in bugzilla. Note: The EPEL field is
              always displayed for packages in the 'rpms' namespace regardless of whether it
              is used in bugzilla or not.            </p>
          </div>
          <div class="modal-footer">
            <button class="btn btn-secondary" type="button" title="Update bugzilla overrides" id="reset_assignees">
               Reset to defaults
            </button>
            <button class="btn btn-primary" type="submit" title="Update bugzilla overrides" id="update_assignees">
               Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>

  <div class="modal fade" id="modal_orphan" tabindex="-1"
          role="dialog" aria-labelledby="Orphan this package" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Orphan package</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
          <div class="modal-body">
            <div class="form-group">
              <label for="reason_menu">Reason</label>
              <select id="reason_menu" class="form-control">

                <option id="lack_of_time_option_button">Lack of time</option>

                <option id="do_not_use_it_option_button">Do not use it anymore</option>

                <option id="unmaintained_option_button">Unmaintained upstream</option>

                <option id="fails_to_build_option_button">Fails to build from source</option>

                <option id="not_fixed_option_button">Important bug not fixed</option>

                <option id="other_option_button">Other</option>
              </select>
            </div>
          <label for="orphan_reason_info">Additional info</label>
          <input title="Additional info for orphaning reason"
                  class="form-control" name="orphan_reason_info" id="orphan_reason_info" value=""/>
          </div>
          <div class="modal-footer">
            <button class="btn btn-primary" type="submit" title="Orphan this package" id="orphan_button">
               Update
            </button>
          </div>
      </div>
    </div>
  </div>

  <script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA">
    window.addEventListener('load', function() {
      set_up_monitoring = function(status){
        var _label = "Disabled"
        if (status === "monitoring") {
          _label = "Monitoring";
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-with-scratch") {
          _label = "Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else {
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye-slash")
        }

        $("#monitoring-label").text(_label);
        $("#monitoring-label").removeClass("fa fa-circle-o-notch fa-spin fa-1x fa-fw");
      }

      $.ajax({
        url: "/_dg/anitya/rpms/plasma-workspace",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          console.log(res);
          set_up_monitoring(res.monitoring)
        }
      });


      $("#reset_assignees").on('click', function(){
        $('#fedora_assignee').val('');
        $('#epel_assignee').val('');
        $("#change_assignees").submit();
        return false;
      });

      $("#change_assignees").on('submit',  function(){
        $('html').css('cursor', 'progress');
        $('#reset_assignees').attr('disabled', true);
        $('#update_assignees').attr('disabled', true);
        $('#update_assignees').text('Updating...');
        $.ajax({
          url: "/_dg/bzoverrides/rpms/plasma-workspace",
          type: 'POST',
          dataType: 'json',
          data: {
             'epel_assignee': $('#epel_assignee').val(),
             'fedora_assignee': $('#fedora_assignee').val()
          },
          success: function(res) {
            $("#fedora_assignee_txt").text(res.fedora_assignee);
            $("#epel_assignee_txt").text(res.epel_assignee);
            $('#modal_assignee').modal('hide');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            $('html').css('cursor', 'default');
            console.log("Successfully changed the bugzilla assignees");
            return false;
          },
          error: function(res) {
            var msg = '';
            if(res.responseJSON.errors){
              msg = ': ' + res.responseJSON.errors.join(', ');
            }
            alert("Unable to update the bugzilla assignee(s)" + msg);
            $('html').css('cursor', 'default');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            return false;
          }
        })
        return false;
      });


      $.ajax({
        url: "/_dg/actived/rpms/plasma-workspace",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          var _btn = $("#take_orphan_button");
          if (!res.active){
            _btn.off("click");
            _btn.click(function(){
                window.open(
                    "https://pagure.io/releng/new_issue?title="
                    + "Unretire rpms/plasma-workspace"
                    + "&amp;template=package_unretiremet");
            });
            _btn.prop( "title", "Package retired - Open a releng ticket to adopt it" );
            _btn.html("Retired");
          }
          _btn.removeClass('disabled');
        }
      });

      $("#orphan_button").click(function(){
        $("#orphan_button").attr("disabled", true);
        $.ajax({
          url: "/_dg/orphan/rpms/plasma-workspace",
          type: 'POST',
          dataType: 'json',
          data: {
            'orphan_reason': $('#reason_menu').val(),
            'orphan_reason_info': $('#orphan_reason_info').val()
          },
          success: function(res) {
            $("#point_of_contact_div").html("Package is currently unmaintained");
            $("#orphan_button").attr("disabled", false);
            $('#modal_orphan').modal('hide');
            $('#orphan-section').html('');
          },
          error: function(res) {
            if (res.responseJSON.errors) {
              alert('Unable to orphan the package: ' + res.responseJSON.errors);
            } else {
              alert('Unable to orphan the package: ' + res.responseJSON.error);
            }
            $("#orphan_button").attr("disabled", false);
          }
        });
      });
    });
  </script>

</nav>    </div>
    <div class="col-10">
  <div class="row mb-1">
    <div class="col-sm-6">
    <h3>
      Files
    </h3>
    </div>

    <div class="col-sm-6">
      <div class="float-right">
          <div class="btn-group">
            <a href="#" class="btn btn-outline-light border-secondary text-dark btn-sm dropdown-toggle"
                    data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    <span class="fa fa-random fa-fw"></span> Branch: <span class="font-weight-bold">rawhide</span>
          </a>
            <div class="dropdown-menu dropdown-menu-right">
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/epel8">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel8</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/epel8-next">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel8-next</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/epel8-playground">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel8-playground</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/epel9">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel9</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/epel9-next">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel9-next</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f20">
                      <span class="fa fa-random fa-fw"></span> <span class="">f20</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f21">
                      <span class="fa fa-random fa-fw"></span> <span class="">f21</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f22">
                      <span class="fa fa-random fa-fw"></span> <span class="">f22</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f23">
                      <span class="fa fa-random fa-fw"></span> <span class="">f23</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f24">
                      <span class="fa fa-random fa-fw"></span> <span class="">f24</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f25">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f26">
                      <span class="fa fa-random fa-fw"></span> <span class="">f26</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f27">
                      <span class="fa fa-random fa-fw"></span> <span class="">f27</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f28">
                      <span class="fa fa-random fa-fw"></span> <span class="">f28</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f29">
                      <span class="fa fa-random fa-fw"></span> <span class="">f29</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f30">
                      <span class="fa fa-random fa-fw"></span> <span class="">f30</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f31">
                      <span class="fa fa-random fa-fw"></span> <span class="">f31</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f32">
                      <span class="fa fa-random fa-fw"></span> <span class="">f32</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f33">
                      <span class="fa fa-random fa-fw"></span> <span class="">f33</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f34">
                      <span class="fa fa-random fa-fw"></span> <span class="">f34</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f35">
                      <span class="fa fa-random fa-fw"></span> <span class="">f35</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/f36">
                      <span class="fa fa-random fa-fw"></span> <span class="">f36</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/plasma-workspace/tree/main">
                      <span class="fa fa-random fa-fw"></span> <span class="">main</span>
                  </a>
                  <a class="dropdown-item pl-1 active" href="/rpms/plasma-workspace/tree/rawhide">
                      <span class="fa fa-random fa-fw"></span> <span class="font-weight-bold">rawhide</span>
                  </a>
            </div>
          </div>
    </div>
  </div>
  </div>
    <div class="card mb-3">
      <div class="card-header">
        <ol class="breadcrumb p-0 bg-transparent mb-0">
          <li class="breadcrumb-item">
            <a href="/rpms/plasma-workspace/tree/rawhide">
              <span class="fa fa-random">
              </span>&nbsp; rawhide
            </a>
          </li>
          <li class="active breadcrumb-item">
            <span class="fa fa-file" data-glyph="">
            </span>&nbsp; disable-toplevel-fixed-positions-on-wayland.patch
          </li>
        </ol>
      </div>

  <div class="card-body p-0">
            <div class="bg-light border text-right pr-2">
                <form class="btn btn-sm" method="POST" name="fork_project"
                    action="/fork_edit/rpms/plasma-workspace/edit/rawhide/f/disable-toplevel-fixed-positions-on-wayland.patch">
                    <button class="btn btn-sm btn-secondary fork_project_btn">
                            Fork and Edit
                    </button>
                    
                </form>

                <a class="btn btn-secondary btn-sm" href="/rpms/plasma-workspace/blob/rawhide/f/disable-toplevel-fixed-positions-on-wayland.patch" title="View as blob">Blob</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/plasma-workspace/blame/disable-toplevel-fixed-positions-on-wayland.patch?identifier=rawhide" title="View git blame">Blame</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/plasma-workspace/history/disable-toplevel-fixed-positions-on-wayland.patch?identifier=rawhide" title="View git log for this file">History</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/plasma-workspace/raw/rawhide/f/disable-toplevel-fixed-positions-on-wayland.patch" title="View as raw">Raw</a>
            </div>

        <pre class="syntaxhighlightblock"><code class="lang-diff">From 1be046edd455969b74417ded6871d25eba8a1112 Mon Sep 17 00:00:00 2001
From: David Edmundson &lt;kde@davidedmundson.co.uk&gt;
Date: Mon, 27 Jun 2022 17:12:09 +0200
Subject: [PATCH] Disable toplevel fixed positions on Wayland

Given that the wayland protocol does not have window positions a recent
change to Qt development branch avoids having a mismatch of positions
and tells client code the window is at the topleft of the screen. This
fixes multiple bugs.

Unfortuantely this breaks a few usages within plasma where a
side-channel sends that absolute positional information over a bespoke
protocol. Currently we proxy this all through QWindow geometry where the
new Qt behaviour will throw is askew.

Whilst our longterm plans are to find future options away from the
plasmashell protocol we may as well have our Qt6
build work given the patch is trivial and harmless.

It also allows for potential backporting if our software is
already secured against issues.
---
 krunner/main.cpp | 2 ++
 shell/main.cpp   | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/krunner/main.cpp b/krunner/main.cpp
index a05402a01..f19382cad 100644
--- a/krunner/main.cpp
+++ b/krunner/main.cpp
@@ -34,6 +34,7 @@ int main(int argc, char **argv)
         QCoreApplication::setAttribute(Qt::AA_DisableHighDpiScaling);
     }

+    qputenv(&#34;QT_WAYLAND_DISABLE_FIXED_POSITIONS&#34;, {});
     const bool qpaVariable = qEnvironmentVariableIsSet(&#34;QT_QPA_PLATFORM&#34;);
     KWorkSpace::detectPlatform(argc, argv);
     QQuickWindow::setDefaultAlphaBuffer(true);
@@ -42,6 +43,7 @@ int main(int argc, char **argv)
         // don&#39;t leak the env variable to processes we start
         qunsetenv(&#34;QT_QPA_PLATFORM&#34;);
     }
+    qunsetenv(&#34;QT_WAYLAND_DISABLE_FIXED_POSITIONS&#34;);
     KLocalizedString::setApplicationDomain(&#34;krunner&#34;);

     //     TODO: Make it a QGuiApplication once we don&#39;t depend on KDELibs4Support
diff --git a/shell/main.cpp b/shell/main.cpp
index 26e4ecede..147483df4 100644
--- a/shell/main.cpp
+++ b/shell/main.cpp
@@ -79,6 +79,7 @@ int main(int argc, char *argv[])

     oldCategoryFilter = QLoggingCategory::installFilter(filterConnectionSyntaxWarning);

+    qputenv(&#34;QT_WAYLAND_DISABLE_FIXED_POSITIONS&#34;, {});
     const bool qpaVariable = qEnvironmentVariableIsSet(&#34;QT_QPA_PLATFORM&#34;);
     KWorkSpace::detectPlatform(argc, argv);
     QApplication app(argc, argv);
@@ -86,6 +87,8 @@ int main(int argc, char *argv[])
         // don&#39;t leak the env variable to processes we start
         qunsetenv(&#34;QT_QPA_PLATFORM&#34;);
     }
+    qunsetenv(&#34;QT_WAYLAND_DISABLE_FIXED_POSITIONS&#34;);
+
     KLocalizedString::setApplicationDomain(&#34;plasmashell&#34;);

     // The executable&#39;s path is added to the library/plugin paths.
--
GitLab
</code></pre>
  </div>
 </div> <!-- end .card-->

</div>
</div>
</div>
    </div>

        <div class="footer pt-4 text-white">
        <div class="container">
            <div class="d-flex align-items-center">
                <div>
                    <div>Powered by <a href="https://pagure.io/pagure" class="notblue">Pagure</a> 5.13.3</div>
                    <div>
                        <a href="https://docs.pagure.org/pagure/usage/index.html" class="notblue">Documentation</a> &bull;
                        <a href="https://pagure.io/pagure/new_issue" class="notblue">File an Issue</a> &bull;
                        <a href="/about">About this Instance</a> &bull;
                        <a href="/ssh_info" class="notblue">SSH Hostkey/Fingerprint</a>
                    </div>
                </div>
                <div class="ml-auto text-right">
                    <div>&copy; Red Hat, Inc. and others.</div>
                </div>
            </div>
        </div>
    </div>


    <script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA" src="/static/vendor/jquery/jquery.min.js?version=5.13.3"></script>

    <script src="/static/vendor/bootstrap/bootstrap.bundle.min.js?version=5.13.3"></script>

    <script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA">
      $('[data-toggle="tooltip"]').tooltip({placement : 'bottom'});
      $(".cancel_btn").click(function() {
        history.back();
      });
    </script>

<script type="text/javascript"  nonce="bYHSDeYpamaMeW4MROg2tpoFA" src="/static/vendor/lazyload/lazyload.min.js?version=5.13.3"></script>

<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA">
window.addEventListener("load", function(event) {
    lazyload();
});
</script>

<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA">
$("#giturl-toggle").on('click', function(event){
  event.stopPropagation();
  $("#giturl-more").toggle();
  $("#giturl-toggle").hide();
})

$(".fork_project_btn").click(function() {
  $('#fork_project').submit();
});

$(".select-on-focus").on("focus", function() {
  $(this).select();
});

</script>

<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA" src="/static/vendor/highlight.js/highlight.pack.js?version=5.13.3"></script>
<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA" src="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.js?version=5.13.3"></script>
<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA" src="/static/vendor/highlight.js/spec.js?version=5.13.3"></script>

<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA">
  $(document).ready(function() {
  $('.fork_project_btn').click($("[name=fork_project]").submit);

  $('pre.syntaxhighlightblock code').each(function(i, block) {
    hljs.highlightBlock(block);
    hljs.lineNumbersBlock(block);
  });

  var cls = "highlighted-line";
  var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
  if (! isNaN(lines[0]))
  {
    for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
      $('#_' + i).parent().parent().addClass(cls);
    }
    setTimeout(function(){
      $("#_" + lines[0]).get(0).scrollIntoView({behavior: "instant", block: "start", inline: "nearest"});
    }, 50);
  }
});
</script>

<script type="text/javascript" nonce="bYHSDeYpamaMeW4MROg2tpoFA">

  function updateHighlight() {
    var cls = "highlighted-line";
    $('.' + cls).removeClass(cls)
    if (location.hash !== '') {
      var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
      for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
        $('[data-line-number=' + i + ']').closest('tr').addClass(cls);
      }
      return lines;
    }
    return [];
  }
  $(window).on('hashchange', updateHighlight);
  var selected = [];
  $("[data-line-number]").click(function (ev) {
    var line = $(this).attr('data-line-number');
    if (ev.shiftKey) {
      selected = selected.slice(-1).concat(line);
    } else {
      selected = [line];
    }

    var hash = '_' + selected[0];
    if (selected.length === 2) {
      hash = '_' + Math.min(selected[0], selected[1]) + '-' + Math.max(selected[0], selected[1]);
    }
    window.location.hash = hash;
    return false;
  });

</script>


</body>
</html>